pkg/volume/host_path/host_path.go

```diff
@@ -245,25 +244,28 @@ type hostPathCreater struct {
 
 // Create for hostPath simply creates a local /tmp/hostpath_pv/%s directory as a new PersistentVolume.
 // This Creater is meant for development and testing only and WILL NOT WORK in a multi-node cluster.
-func (r *hostPathCreater) Create() (*api.PersistentVolume, error) {
-	fullpath := fmt.Sprintf("/tmp/hostpath_pv/%s", util.NewUUID())
-	err := os.MkdirAll(fullpath, 0750)
-	if err != nil {
-		return nil, err
+func (r *hostPathCreater) Provision(pv *api.PersistentVolume) error {
+	if pv.Spec.HostPath == nil {
+		return fmt.Errorf("pv.Spec.HostPath cannot be nil")
 	}
+	return os.MkdirAll(pv.Spec.HostPath.Path, 0750)
+}
 
+func (r *hostPathCreater) NewPersistentVolumeTemplate() (*api.PersistentVolume, error) {
+	fullpath := fmt.Sprintf("/tmp/hostpath_pv/%s", util.NewUUID())
 	return &api.PersistentVolume{
 		ObjectMeta: api.ObjectMeta{
 			GenerateName: "pv-hostpath-",
 			Labels: map[string]string{
-				"createdby": "hostpath dynamic provisioner",
+				"createdby": "hostpath-dynamic-provisioner",
```

Comment:
1) Shouldn't this be an annotation?  It's not part of the identity - is it actively used to select PVs?

2) This should have a kubernetes.io/ prefix - we must NEVER use naked annotations/labels - those are reserved for users.